### PR TITLE
fix: Lighthouse a11y improvements + fixes for Lighthouse SEO errors

### LINKED ISSUE #2790: enh: accessibility
#277

### PR DESCRIPTION
**Pull request by sebdanielsson**
# Changelog Entry

### Description

This PR addresses lots of low-hanging fruit fixes to greatly improve the Lighthouse score. After these fixes:

-  No errors are reported for the SEO tests and it is now scored at 66, up from 50. Only way to improve it is to allow search engines to index which is probably unwanted for most people.
-  Accessibility score has improved from 71 to 100 by improving contrast, adding aria label for some buttons, and increasing the size on one button.

### Fixed

- The inclusion of a robot.txt file with Disallow All addresses a Lighthouse SEO *error*, preventing content indexing by search engines.
- A meta description has been added, addressing a Lighthouse recommendation and positively impacting the SEO score.
- Missing aria labels on buttons have been added, enhancing the Lighthouse accessibility test results.
- The "More Models" button has been adjusted, including margin and size increases, to improve the Lighthouse score and align with WCAG.
- The maximum-scale setting has been updated to 2 to comply with Lighthouse and WCAG zoom functionality recommendations.
- Text color adjustments have been made to enhance contrast, improving the Lighthouse accessibility score and aligning with WCAG guidelines.

---

### Additional Information

- Partly fixes #2790. There's still some more accessibility fixes to do on the admin panel. But the login page and main page should be greatly improved.

### Screenshots or Videos

image

image

### CODE REVIEW

**Comment by tjbck** [comment]
I appreciate the PR, and PRs for aria-labels are generally welcomed but please do not change the styling as we won't be merging them, Thanks!

File: src/app.html [hunk 1]
```diff
@@ -4,8 +4,13 @@
 		<meta charset="utf-8" />
 		<link rel="icon" href="%sveltekit.assets%/favicon.png" />
 		<link rel="manifest" href="%sveltekit.assets%/manifest.json" crossorigin="use-credentials" />
-		<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1" />
+		<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2" />
 		<meta name="robots" content="noindex,nofollow" />
+		<meta
```
Comment (tjbck) [inline src/app.html:9]: Should just say "Open WebUI"

File: src/app.html [hunk 2]
```diff
@@ -4,8 +4,13 @@
 		<meta charset="utf-8" />
 		<link rel="icon" href="%sveltekit.assets%/favicon.png" />
 		<link rel="manifest" href="%sveltekit.assets%/manifest.json" crossorigin="use-credentials" />
-		<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1" />
+		<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2" />
```
Comment (tjbck) [inline src/app.html:7]: Shouldn't be changed.

Comment (sebdanielsson) [reply src/app.html:7]: What's the reasoning behind this?

https://dequeuniversity.com/rules/axe/4.9/meta-viewport

**Comment by sebdanielsson** [comment]
> I appreciate the PR, and PRs for aria-labels are generally welcomed but please do not change the styling as we won't be merging them, Thanks!

Hi, thank you for the quick response!

Any reason why styling changes won't be merged? Is it part of a bigger overhaul to address the remaining issues?

**Pull request closed without merging**